pyface/tests/test_resource_manager.py

```diff
@@ -47,3 +51,41 @@ def __len__(self):
         resource_manager = ResourceManager()
         img_ref = resource_manager.locate_image("core.png", sequence)
         self.assertEqual(IMAGE_PATH, img_ref.filename)
+
+    def test_locate_image_with_module(self):
+        # images/close.png is included in pyface package data.
+        # ResourceManager can find it via package resources
+        resource_manager = ResourceManager()
+        image_ref = resource_manager.locate_image("close.png", [pyface])
+        self.assertGreater(len(image_ref.data), 0)
+
+    def test_locate_image_with_module_missing_file(self):
+        # The required image is not found, locate_image should return None.
+        resource_manager = ResourceManager()
+        image_ref = resource_manager.locate_image(
+            "does_not_exist.png", [pyface]
+        )
+        self.assertIsNone(image_ref)
+
+    def test_locate_image_with_name_being_dunder_main(self):
+        # When a module is __main__, we will fall back to use __file__
+
+        # given a module from which there is an image in the same folder
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            shutil.copyfile(IMAGE_PATH, os.path.join(tmp_dir, "random.png"))
+            py_filepath = os.path.join(tmp_dir, "tmp.py")
+            with open(py_filepath, "w", encoding="utf-8"):
+                pass
```

Comment:
What exactly are these two lines needed for?

Reply:
This creates an empty file, because the content of the file does not matter for this test. Will add a comment.

Reply:
ah i see now, that makes sense thanks